perun-engine/src/main/java/cz/metacentrum/perun/engine/jms/MessageReceiver.java

```diff
@@ -85,6 +85,11 @@ public void run() {
 					waitTime = waitTime + 5000;
 					log.error("Queue doesn't exist yet. We gonna wait a bit ({} s) and try it again.",
 							(waitTime / 1000), e);
+					try {
+						Thread.sleep(waitTime);
+					} catch (InterruptedException interrupted) {
+						log.error(interrupted.toString(), interrupted);
+					}
```

Comment:
It might be also worth adding an inline comment describing the reason for this wait.

Reply:
Inline comment added before try-catch block.